### PR TITLE
Plumbox Launcher and CLI

### PR DESCRIPTION
**Pull request by Talina06**

### CODE REVIEW

File: src/main/java/in/oogway/plumbox/cli/Localstorage.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package in.oogway.plumbox.cli;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+
+public class Localstorage extends StorageDriver {
+    YamlHandler yamlHandler = YamlHandler.createYamlHandler();
```
Comment (meson10) [inline src/main/java/in/oogway/plumbox/cli/Localstorage.java]: The format is JSON now right? It cannot be YAML. If the Local implementation is not used, please delete it.

File: src/main/java/in/oogway/plumbox/cli/Plumbox.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            Map<String, Object> data = yamlHandler.readYAML(this.CONF_PATH);
+            this.PLUMBOX_BASE_DIR = (String) data.get("path");
+            this.filesystem = (String) data.get("filesystem");
+
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        if (this.filesystem.equals("LOCAL")) {
+            this.storageDriver = new Localstorage();
```
Comment (meson10) [inline src/main/java/in/oogway/plumbox/cli/Plumbox.java]: LocalStorage stores as YAML whereas Redis sores JSON. Keep them as same or drop the one that is not being used.

File: src/main/resources/input_source_file.json [hunk 3]
```diff
@@ -0,0 +1 @@
...06,"longitude":73.850043,"timestamp":1516399872},{"busID":1,"boxID":1,"latitude":18.531968,"longitude":73.850226,"timestamp":1516395912},{"busID":1,"boxID":1,"latitude":18.529099,"longitude":73.824069,"timestamp":1516389312},{"busID":1,"boxID":1,"latitude":18.529099,"longitude":73.824069,"timestamp":1516387512},{"busID":1,"boxID":1,"latitude":18.531968,"longitude":73.850226,"timestamp":1516472412},{"busID":1,"boxID":1,"latitude":18.531398,"longitude":73.852115,"timestamp":1516470132},{"busID":1,"boxID":1,"latitude":18.53207,"longitude":73.853381,"timestamp":1516314312},{"busID":1,"boxID":1,"latitude":18.54865,"longitude":73.845699,"timestamp":1516309512},{"busID":1,"boxID":1,"latitude":18.531806,"longitude":73.845699,"timestamp":1516384812},{"busID":1,"boxID":1,"latitude":18.527675,"longitude":73.841708,"timestamp":1516224852},{"busID":1,"boxID":1,"latitude":18.527675,"longitude":73.841708,"timestamp":1516222812},{"busID":1,"boxID":1,"latitude":18.527675,"longitude":73.841708,"timestamp":1516221252},{"busID":1,"boxID":1,"latitude":18.531663,"longitude":73.851299,"timestamp":1516217652},{"busID":1,"boxID":1,"latitude":18.5318058,"longitude":73.85004320000007,"timestamp":1516296852}]
```
Comment (meson10) [inline src/main/resources/input_source_file.json]: This should be in src/test and not in main.

File: src/main/java/in/oogway/plumbox/launcher/transformer/Transformer.java [hunk 4]
```diff
@@ -1,5 +1,8 @@
 package in.oogway.plumbox.launcher.transformer;
 
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+
 public interface Transformer {
-    void run();
+    Dataset<Row> run(Dataset<Row> events);
```
Comment (meson10) [inline src/main/java/in/oogway/plumbox/launcher/transformer/Transformer.java]: rows and not events.

File: src/main/resources/input_source_file.json (hunk 3 above)
Comment (meson10) [reply src/main/resources/input_source_file.json]: Also, this doesn't belong to this project.

File: src/main/java/in/oogway/plumbox/launcher/storage/RedisStorage.java [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
 import java.nio.charset.Charset;
 
 /*
 *   @author talina06 on 2/7/18
 */
 public interface RedisStorage  extends StorageDriver {
+
+    default void redisServer() {
+        String address = Config.getDirPath("redis_server_address");
+        Config.jedis = new Jedis(address);
```
Comment (meson10) [inline src/main/java/in/oogway/plumbox/launcher/storage/RedisStorage.java]: Not sure if this deserves to go to Config. It should be a static/private singleton.

File: src/main/java/in/oogway/plumbox/launcher/storage/JSONParser.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import java.io.IOException;
+import java.io.Reader;
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+*   @author talina06 on 2/15/18
+*/
+
+public interface JSONParser extends RedisStorage{
```
Comment (meson10) [inline src/main/java/in/oogway/plumbox/launcher/storage/JSONParser.java]: JSONParser is too ambiguous a name.

File: src/main/java/in/oogway/plumbox/launcher/storage/JSONParser.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    default Map loadContent(String id)
+            throws IllegalAccessException, InstantiationException, ClassNotFoundException, IOException {
+        byte[] bytes = read(id);
+        Reader fileReader = getFileReader(bytes);
```
Comment (meson10) [inline src/main/java/in/oogway/plumbox/launcher/storage/JSONParser.java]: I am not sure if converting to file and loading is the best way to load content from JSON.

File: src/main/java/in/oogway/plumbox/launcher/source/Source.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import in.oogway.plumbox.launcher.config.Config;
+import in.oogway.plumbox.launcher.storage.JSONParser;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Map;
+
+public class Source implements JSONParser{
```
Comment (meson10) [inline src/main/java/in/oogway/plumbox/launcher/source/Source.java]: Use some generic here. implements JSONParser < KnownSourceSchema >

**Review by meson10 — CHANGES_REQUESTED** [review]

**Comment by Talina06** [comment]
Work_branch already merged.

**Pull request closed without merging**